src/config.c

```diff
@@ -711,6 +711,50 @@ int parseCommandLine(int argc,char **argv,ParodusCfg * cfg)
     return 0;
 }
 
+void free_cfg(ParodusCfg *cfg)
+{
+    if (cfg->mtls_client_cert_path != NULL )
```

Comment:
Can you add cfg NULL check and assign it to NULL at the end once elements are freed .

Reply:
cfg I traced back is pointer to static global struct variable so ideally we should not assign it NULL 
https://github.com/xmidt-org/parodus/blob/master/src/config.c#L44

Reply:
Added NULL check but not able to run guardrails again to confirm